### PR TITLE
Add examples

### PR DESCRIPTION
**Pull request by BruceMacD**
Add some examples to match Python:
https://github.com/jmorganca/ollama-python/tree/main/examples

How to run:
```
npm install --global tsx
tsx examples/example-name/example.ts
```

### CODE REVIEW

File: examples/generate/generate.ts [hunk 1]
```diff
@@ -0,0 +1,11 @@
+import ollama from 'ollama'
+
+async function main(): Promise<void> {
+  const respose = await ollama.generate({
+    model: 'mistral',
+    prompt: 'Why is the sky blue?',
+  })
+  console.log(respose.response)
```
Comment (mxyng) [inline examples/generate/generate.ts:8]:
```suggestion
  console.log(response.response)
```

File: examples/fill-in-middle/fill.ts [hunk 2]
```diff
@@ -0,0 +1,24 @@
+import ollama, { Message } from 'ollama'
+
+async function main(): Promise<void> {
+  const prefix = `def remove_non_ascii(s: str) -> str:
+"""
+`
+  const suffix = `
+return result
+`
+  const respose = await ollama.generate({
```
Comment (mxyng) [inline examples/fill-in-middle/fill.ts:10]:
```suggestion
  const response = await ollama.generate({
```

File: examples/async-chat-stream/chat.ts [hunk 3]
```diff
@@ -0,0 +1,102 @@
+import { spawn, ChildProcessWithoutNullStreams, execSync } from 'child_process'
```
Comment (jmorganca) [inline examples/async-chat-stream/chat.ts:1]: Each of these should be a few lines of code – copy and paste-able

File: examples/fill-in-middle/fill.ts [hunk 4]
```diff
@@ -0,0 +1,20 @@
+import ollama, { Message } from 'ollama'
```
Comment (jmorganca) [inline examples/fill-in-middle/fill.ts:1]: This one is great

File: examples/multimodal/multimodal.ts [hunk 5]
```diff
@@ -0,0 +1,40 @@
+import https from 'https'
```
Comment (jmorganca) [inline examples/multimodal/multimodal.ts:1]: Let's just load a file from disk in the repo (e.g. the watermelon bike or astronaut on horse) and show – the key idea here isn't to show a full app, but something quick to understand and copy and paste

File: examples/pull-progress/pull.ts [hunk 6]
```diff
@@ -0,0 +1,25 @@
+import ollama from 'ollama'
```
Comment (jmorganca) [inline examples/pull-progress/pull.ts:1]: great!

File: examples/async-chat-stream/chat.ts (hunk 3 above)
Comment (BruceMacD) [reply examples/async-chat-stream/chat.ts]: removed this example

File: examples/multimodal/multimodal.ts (hunk 5 above)
Comment (BruceMacD) [reply examples/multimodal/multimodal.ts:1]: Used a local cat image instead

File: examples/pull-progress/pull.ts [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+
+const model = 'falcon'
```
Comment (jmorganca) [inline examples/pull-progress/pull.ts:3]: let's switch this to `mistral`

```suggestion
const model = 'mistral'
```

**Pull request merged**